### DISCUSSION TITLE
Connecting to a Postgres DB timed out

### DISCUSSION THREAD

**Discussion by sr1dar** [community/community#30056] [Actions]
![image](https://user-images.githubusercontent.com/100928638/185596410-3511174b-3184-4de2-96b1-ffe43268fe81.png)

I'm trying to connect to a RDS postgres database by running python shell commands and the connection times out every time.
I'm able to connect to the DB when I run the code on my personal IDE, So DB must be properly configured.
I just want to write queries using python..

Python code:
```
import os
import psycopg2
from datetime import datetime  
now = datetime.now()

run_id=${{ github.run_id }}
env_file="string"
arr=env_file.split('/') #SQL File name
file_name=arr[len(arr)-1]
output_dict={ }
current_time = now.strftime("%Y/%m/%d %H:%M:%S")

try:
    connection = psycopg2.connect(user="",
                                    password="",
                                    host="",
                                    port="",
                                    database="")
    cursor = connection.cursor()
    for k,v in output_dict.items():
        str=''
        for i in v["comments"]:
            str+=i+'\n'
        #print(str)
        postgres_insert_query = """ INSERT INTO """
        record_to_insert = (run_id, file_name, k,v['flag'],str,current_time)
        cursor.execute(postgres_insert_query, record_to_insert)
        connection.commit()
        count = cursor.rowcount
        print(count, "Record inserted successfully inserted")

except (Exception, psycopg2.Error) as error:
    print("Failed to insert record ", error)

finally:
    # closing database connection.
    if connection:
        cursor.close()
        connection.close()
        print("PostgreSQL connection is closed")
```

Any response would be appreciated, Thanks.

**Comment by airtower-luna** [comment]
The code you quoted is not the line mentioned in the exception. You'll need to show a lot more context for anyone to be able to help.

When posting code please remember to [use a code block](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks) (enclosed in triple backticks) so newlines and indentation are preserved. That's critical for a language like Python where indentation is part of the syntax.

**Reply by sr1dar** [reply]
> The code you quoted is not the line mentioned in the exception. You'll need to show a lot more context for anyone to be able to help.
> 
> When posting code please remember to [use a code block](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks) (enclosed in triple backticks) so newlines and indentation are preserved. That's critical for a language like Python where indentation is part of the syntax.

Hi sorry about that, Updated with the entire code

**Reply by sr1dar** [reply]
I even tried connecting with asyncpg, Works on my IDE but not on the github runner

**Reply by airtower-luna** [reply]
Okay, with that code I can tell that the `psycopg2.connect()` call fails, so the `connection` variable is never assigned, and that's why reading it in the `finally` block crashes.

> I'm able to connect to the DB when I run the code on my personal IDE, So DB must be properly configured.

A database working on your system doesn't mean it works on Actions, so that's the first thing I'd check. The other thing I'd check is exactly where `psycopg2.connect()` tries to connect if its parameters are all empty. If it works on your machine it's probably localhost and a default port, but such details matter.
